app/models/foreman_openscap/arf_report.rb

```diff
@@ -192,5 +201,22 @@ def self.references_links(references)
       end
       html_links.join(', ')
     end
+
+    def self.check_msg_for_changes(msg, incoming_data)
+      mapping = { :severity => :severity,
+                  :description => :description,
+                  :rationale => :rationale,
+                  :scap_references => :references,
+                  :value => :title }
+      mapping.each do |msg_attr, data_key|
+        msg.public_send("#{msg_attr}=".to_sym, incoming_data[data_key]) unless msg.public_send(msg_attr) == incoming_data[data_key]
```

Comment:
why does it check for the value change before assignment? wouldn't it be checked 2 lines below by `msg.changed?` I'd also prefer to avoid this meta-programming, using `public_send` is harder to read, especially with using constructions like `"#{msg_attr}=".to_sym`

if I compare it to
```ruby
msg.severity = incoming_data['severity']
msg.description = incoming_data['description']
msg.rationale = incoming_data['rationale']
msg.scap_references = incoming_data['references']
msg.value = incoming_data['title']
```
it's less code (even if I added `unless` condition to every line) and it's immediately clear what it does